### PR TITLE
[DSPDC-1817] Refactor common transformation record grouping code

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1817)
There is much repeated code in our transformation pipeline builders, specifically with respect to reading JSON and transforming to `RawRecord` instances.

## This PR
* Creates a `TransformationHelper` class with necessary grouping semantics
* Adds a test around the methods

## Checklist
- [ ] Documentation has been updated as needed.

### CODE REVIEW

**Review by quazi-h — APPROVED** [review]
lgtm! This will make it much easier to put up new PipelineBuilders.

I was recently wondering whether we should be grouping the records for all of the pipelines by  redcap_event_name. HLES, Sample, and EOLS are all pulling a single arm (baseline) anyways, so grouping by the redcap event should produce the same output. In particular I was thinking about what changes we will have to make when we have HLES follow up surveys which I think would introduce new arms for us to pull - something Matt has mentioned in the past but I suppose we can handle that pretty easily by just switching the specific function being called when that comes.

File: dap-etl/transformation/src/test/scala/org/broadinstitute/monster/dap/common/TransformationHelperSpec.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.monster.dap.common
+
+import com.spotify.scio.testing.PipelineSpec
+import org.apache.beam.sdk.options.{PipelineOptions, PipelineOptionsFactory}
+import org.scalatest.matchers.should.Matchers
+
+class TransformationHelperSpec extends PipelineSpec with Matchers {
```
Comment (quazi-broad) [inline dap-etl/transformation/src/test/scala/org/broadinstitute/monster/dap/common/TransformationHelperSpec.scala:7]: Actually, I think the spec is failing.

Comment (aherbst-broad) [reply dap-etl/transformation/src/test/scala/org/broadinstitute/monster/dap/common/TransformationHelperSpec.scala:7]: Ah, forgot to add the needed fixture data file.

**Comment by aherbst-broad** [comment]
@quazi-broad great point; I think we'll likely want a record per arm globally as you say. Is there any reason _not_ to make that change now?

File: dap-etl/transformation/src/test/resources/org/broadinstitute/monster/dap/common/transformation_helper/records/records.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{"record":"11111","redcap_event_name":"jan2020_arm_1","field_name":"study_id","value":"11111"}
```
Comment (raaidbroad) [inline dap-etl/transformation/src/test/resources/org/broadinstitute/monster/dap/common/transformation_helper/records/records.json:1]: I am having trouble seeing where this is used 👀

File: dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/common/TransformationHelper.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          RawRecord(id.toLong, fields + ("redcap_event_name" -> Array(eventName)))
+      }
+  }
+
+  private def readJsonLists(ctx: ScioContext, inputPrefix: String): SCollection[Msg] = {
+    StorageIO
+      .readJsonLists(
+        ctx,
+        "Raw Records",
+        s"${inputPrefix}/records/*.json"
```
Comment (raaidbroad) [inline dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/common/TransformationHelper.scala:36]: is this where it is used?

Comment (aherbst-broad) [reply dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/common/TransformationHelper.scala:36]: @raaidbroad correct. [This ](https://github.com/DataBiosphere/dog-aging-ingest/blob/fd86f096d386d971d6bdfcaee6fbea439de77006/dap-etl/transformation/src/test/scala/org/broadinstitute/monster/dap/common/TransformationHelperSpec.scala#L11) gets the relevant path in the `resources` folder and we by convention always look for a `records/` subfolder underneath.

File: dap-etl/transformation/src/test/resources/org/broadinstitute/monster/dap/common/transformation_helper/records/records.json (hunk 2 above)
Comment (aherbst-broad) [reply dap-etl/transformation/src/test/resources/org/broadinstitute/monster/dap/common/transformation_helper/records/records.json:1]: I commented below, but here is where it's [used](https://github.com/DataBiosphere/dog-aging-ingest/blob/fd86f096d386d971d6bdfcaee6fbea439de77006/dap-etl/transformation/src/test/scala/org/broadinstitute/monster/dap/common/TransformationHelperSpec.scala#L11)

**Review by raaidbroad — APPROVED** [review]

**Pull request merged**